regex_checker.py

```
@@ -0,0 +1,114 @@
+                json.dump(captured_groups, f_captured, indent=2, ensure_ascii=False)
+
+    except FileNotFoundError as e:
+        print(f"ファイルが見つかりません: {e}", file=sys.stderr)
+        sys.exit(1)
+    except IOError as e:
+        print(f"ファイルの読み書きエラー: {e}", file=sys.stderr)
+        sys.exit(1)
+
+    return total_lines, matched_count, unmatched_count, captured_groups
+
+def main():
```

Ruff (0.8.2)

Missing return type annotation for public function `main`

Add return type annotation: `None`

(ANN201)